rplugin/python3/denite/source/workspaceSymbol.py

```diff
@@ -17,28 +20,33 @@ def __init__(self, vim):
         self.name = 'workspaceSymbol'
         self.kind = 'file'
 
-    def convert_to_candidates(self, symbols: List[Dict]) -> List[Dict]:
-        candidates = []
-        pwd = self.vim.funcs.getcwd()
-        for sb in symbols:
-            name = sb["name"]
-            uri = sb["location"]["uri"]
-            filepath = uri_to_path(uri)
-            relpath = path.relpath(filepath, pwd)
-            start = sb["location"]["range"]["start"]
-            line = start["line"] + 1
-            character = start["character"] + 1
-            candidates.append({
-                "word": "{}:{}:{}:\t{}".format(
-                    relpath, line, character, name),
-                "action__path": filepath,
-                "action__line": line,
-                "action__col": character,
-            })
-
-        return candidates
+    def highlight(self):
+        highlight_setup(self, SYMBOL_CANDIDATE_HIGHLIGHT_SYNTAX)
 
     def gather_candidates(self, context):
+        context['is_interactive'] = True
+        prefix = context['input']
+        bufnr = context['bufnr']
+
+        cached_prefix = context.get('__cached_prefix')
+        if cached_prefix is not None and prefix.startswith(cached_prefix):
+            return context['__cached_candidates']
+
+        current_buffer = self.vim.current.buffer.number
+        if current_buffer != bufnr:
+            self.vim.command("tabedit %")
+            self.vim.command(
+                "execute 'noautocmd keepalt buffer' {}".format(bufnr))
         result = self.vim.funcs.LanguageClient_runSync(
-            'LanguageClient#workspace_symbol', '', {}) or []
-        return self.convert_to_candidates(result)
+            'LanguageClient#workspace_symbol', prefix, {}) or []
+        if current_buffer != bufnr:
+            self.vim.command("tabclose")
+
+        candidates = convert_symbols_to_candidates(
+            result,
+            pwd=self.vim.funcs.getcwd())
+        if len(result) < self.max_candidates:
```

Comment:
Where is this max_candidates come from?

Reply:
from [Base](https://github.com/Shougo/denite.nvim/blob/cb1daf74c51b1670ee3914c95f85ba3852525e17/rplugin/python3/denite/source/base.py#L19)

Reply:
I see.

Although, from the documentation and implementation, it seems like max_candidates are only applied to candidates filtering,
- https://github.com/Shougo/denite.nvim/blob/a59c3e2f50452e0ddd1988e921624f6cfb47807d/doc/denite.txt#L1957
- https://github.com/Shougo/denite.nvim/blob/0caf719e590bd53c165cdb3f8fdb752f0c336bba/rplugin/python3/denite/denite.py#L161

Reply:
It appears you are correct. It just so happens that ccls just chose the same number as the limit of it's results.

Looking at the protocol I don't see a way for the server to notify the client that the result set is not complete.

I will change it to always ask the server no matter how many results.